xarray/tests/test_nep18.py

```diff
@@ -0,0 +1,186 @@
+import pickle
+from collections import OrderedDict
+from contextlib import suppress
+from distutils.version import LooseVersion
+from textwrap import dedent
+
+import numpy as np
+import pandas as pd
+import pytest
+
+import xarray as xr
+import xarray.ufuncs as xu
+from xarray import DataArray, Dataset, Variable
+from xarray.tests import mock
+from xarray.core.npcompat import IS_NEP18_ACTIVE
+
+from . import (
+    assert_allclose, assert_array_equal, assert_equal, assert_frame_equal,
+    assert_identical, raises_regex)
+
+
+def xfail(param):
+    return pytest.param(param, marks=pytest.mark.xfail)
+
+
+sparse = pytest.importorskip('sparse')
+from sparse.utils import assert_eq
+
+
+@pytest.mark.parametrize("func", [
+    lambda s: s[5:15, 5:15],
+    lambda s: s + 1,
+    lambda s: s + s,
+    lambda s: 2 * s,
+    lambda s: np.sin(s),
+    lambda s: s.sum(skipna=True, axis=0),
+    lambda s: s.sum(skipna=True, axis=1),
+    lambda s: s.sum(skipna=True, axis=0, keepdims=True),
+    xfail(lambda s: s.sum(skipna=False, axis=0)),
+    xfail(lambda s: s.sum(skipna=True)),
+    xfail(lambda s: s.groupby('dim_0').sum())
+])
+@pytest.mark.skipif(not IS_NEP18_ACTIVE,
+                    reason="NUMPY_EXPERIMENTAL_ARRAY_FUNCTION is not enabled")
+def test_sparse(func):
+
+    S = sparse.random((10, 10), random_state=0, density=0.2)
+    A = xr.DataArray(S)
+    assert isinstance(func(A).data, sparse.SparseArray)
+
+    x = xr.Variable('x', sparse.COO.from_numpy([1, 0, 2, 0, 0, 3, 0, 4]))
+    y = xr.Variable('y', sparse.COO.from_numpy([0, 4, 0, 3, 2, 0, 0, 1]))
+    data = sparse.random((8, 8), random_state=0, density=0.2)
+    A = xr.DataArray(data, dims=['x', 'y'])
+    A.coords['xx'] = x
+    A.coords['yy'] = y
+    assert isinstance(func(A).data, sparse.SparseArray)
+
+
+class TestSparseVariable:
```

Comment:
An aside (don't let this slow you down): these are tests that we could run across any eligible array, including the current numpy & dask arrays

You could pull tests from the current dask tests if that would be a shortcut to building up this test suite from scratch

We could do that by inheriting this class, or having `var` as a fixture

Reply:
Thanks @max-sixty. I did copy them over from the dask tests. I'm still figuring out which ones are transferable to this use case since I'm not a regular xarray user myself (yet) :)

Reply:
Great! LMK if I can be helpful at all. Excited about the PR!